File: Solutions/Problem1021/Gopi/Problem1021RemoveOutermostParentheses.js
var removeOuterParentheses = function(s) {
    
    let stack = [];
    let result = "";
    for(let i=0; i<s.length; i++) {

        if(stack.length >= 1) {

            if(stack.length == 1 && s.charAt(i) != ')') {
                result = result+ s.charAt(i);
            } else if(stack.length > 1) {
                result = result + s.charAt(i);
            }
        }
        
        if(s.charAt(i) == '(') {
            stack.push(s.charAt(i));
        } 
        else if(s.charAt(i) == ')' ) {
            stack.pop();
        }
    }
    return result;
};

removeOuterParentheses("(()())(())");

File: Solutions/Problem1021/Gopi/Problem1021RemoveOutermostParentheses.java
package Solutions.Problem1021.Gopi;

import java.util.Stack;

public class Problem1021RemoveOutermostParentheses {
 
    public String removeOuterParentheses(String s) {
        
        StringBuilder resultString = new StringBuilder();

        Stack<Character> stackStore = new Stack<Character>();
        char[] store = s.toCharArray();
        
        for(int i=0; i< store.length; i++) {

            if((stackStore.size() >= 1)){
                if(stackStore.size() == 1 && store[i] != ')') {
                    resultString.append(store[i]); 
                }
                else if(stackStore.size() > 1) {
                    resultString.append(store[i]);
                }
                
            }

            if(store[i] == '(') stackStore.push(store[i]);
            else if(store[i] == ')') stackStore.pop();
        }

        return resultString.toString();
    }

    public static void main(String[] args) {
        Problem1021RemoveOutermostParentheses obj = new Problem1021RemoveOutermostParentheses();

        String ans = obj.removeOuterParentheses("()()");

        System.out.println(ans);

    }
}
